react/src/views/CreateBox/CreateBox.test.js

```diff
@@ -0,0 +1,270 @@
+        expect(component.queryByTestId("createBoxForm")).toBeTruthy();
+        expect(component.queryByTestId("createdBox")).toBeNull();
+        expect(component.queryByTestId("loadingState")).toBeNull();
+        expect(component.queryByTestId("errorState")).toBeNull();
+      }, 5000);
+    });
+  });
+});
+
+describe("Network error after submission", () => {
+  let component = null;
+  beforeEach(() => {
+    const history = createMemoryHistory();
+    const state = { qr: "barcode=387b0f0f5e62cebcafd48383035a92a" };
+    history.push("/create-box", state);
+
+    component = render(
+      <MockedProvider
+        mocks={[
+          {
+            request: {
+              query: CREATE_BOX,
+              variables: {
+                productId: 2,
+                items: 2,
+                locationId: 2,
+                comments: "",
+                sizeId: 2,
+                qrBarcode: "387b0f0f5e62cebcafd48383035a92a",
+              },
+            },
+            error: new Error("An error occurred"),
+          },
+        ]}
+        addTypename={false}
+      >
+        <Router history={history}>
+          <CreateBox />
+        </Router>
+      </MockedProvider>,
+    );
+  });
+
+  afterEach(cleanup);
+
+  it("renders `Error :( Please try again` when there is a network error", async () => {
+    const submitBtn = component.getByRole("button", { name: /do the mutation/i });
+
+    fireEvent.click(submitBtn);
+
+    await waitFor(() => {
+      expect(component.getByText("Error :( Please try again")).toBeInTheDocument();
+    });
+  });
+});
+
+describe("GraphQL error after submission", () => {
+  let component = null;
+  beforeEach(() => {
+    const history = createMemoryHistory();
+    const state = { qr: "barcode=387b0f0f5e62cebcafd48383035a92a" };
+    history.push("/create-box", state);
+
+    component = render(
+      <MockedProvider
+        mocks={[
+          {
+            request: {
+              query: CREATE_BOX,
+              variables: {
+                productId: 2,
+                items: 2,
+                locationId: 2,
+                comments: "",
+                sizeId: 2,
+                qrBarcode: "387b0f0f5e62cebcafd48383035a92a",
+              },
+            },
+            result: {
+              errors: [new GraphQLError("Error!")],
+            },
+          },
+        ]}
+        addTypename={false}
+      >
+        <Router history={history}>
+          <CreateBox />
+        </Router>
+      </MockedProvider>,
+    );
+  });
+
+  afterEach(cleanup);
+
+  it("renders `Error :( Please try again` when there is a GraphQL error", async () => {
+    const submitBtn = component.getByRole("button", { name: /do the mutation/i });
+
+    fireEvent.click(submitBtn);
+
+    await waitFor(() => {
+      expect(component.getByText("Error :( Please try again")).toBeInTheDocument();
+    });
+  });
+});
+
+// Loading state is a work in progress
```

Comment:
please add a comment why it did not work

Reply:
Is it enough if I include this link?: https://github.com/apollographql/apollo-client/issues/6383
If not, I can provide more details.